script/tool/lib/src/format_command.dart

```diff
@@ -120,6 +121,15 @@ class FormatCommand extends PackageLoopingCommand {
       relativeTo: package.directory,
     );
     if (getBoolArg(_dartArg)) {
+      // Ensure that .dart_tool exists, since without it `dart` doesn't know
+      // the lanugage version, so the formatter may give different output.
+      if (!package.directory.childDirectory('.dart_tool').existsSync()) {
```

Comment:
Should we warn through the console in case this step needs to be performed?

That would let us know if something like `flutter clean` gets added by mistake?

Reply:
We pass stdout/stderr through in `runPubGet`, so it's super obvious in the console when this is happening.